rusoto/credential/src/profile.rs

```diff
@@ -307,107 +305,51 @@ fn parse_config_file(file_path: &Path) -> Option<HashMap<String, HashMap<String,
 }
 
 /// Parses a Credentials file into a Map of <`ProfileName`, `AwsCredentials`>
-fn parse_credentials_file(
-    file_path: &Path,
-) -> Result<HashMap<String, AwsCredentials>, CredentialsError> {
-    match fs::metadata(file_path) {
-        Err(_) => {
-            return Err(CredentialsError::new(format!(
-                "Couldn't stat credentials file: [ {:?} ]. Non existant, or no permission.",
-                file_path
+fn parse_credentials_file<P>(
+    file_path: P,
+) -> Result<HashMap<String, AwsCredentials>, CredentialsError>
+where
+    P: AsRef<Path>,
+{
+    let profiles: HashMap<_, _> = parse_config_file(&file_path)
+        .map(|data| {
+            Ok(data
+                .into_iter()
+                .filter_map(|(profile, properties)| {
+                    if let (Some(key), Some(secret)) = (
+                        properties.get("aws_access_key_id"),
+                        properties.get("aws_secret_access_key"),
+                    ) {
+                        Some((
+                            profile,
+                            AwsCredentials::new(
+                                key,
+                                secret,
+                                properties
+                                    .get("aws_session_token")
```

Comment:
This tries to stick close the the previous behavior where it's one of the other but not both. 

Also wondering if there should be some model representation of a `Profile` object that adds more structure to this data. The AWS docs define a [much richer set of fields](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-global)